pkg/dns/dns.go

```diff
@@ -420,8 +420,20 @@ func (kd *KubeDNS) newHeadlessService(service *kapi.Service) error {
 func (kd *KubeDNS) Records(name string, exact bool) (retval []skymsg.Service, err error) {
 	glog.Infof("Received DNS Request:%s, exact:%v", name, exact)
 	trimmed := strings.TrimRight(name, ".")
-	segments := strings.Split(trimmed, ".")
-	path := reverseArray(segments)
+	path := strings.Split(trimmed, ".")
+	isFederationQuery := false
+	federationPath := []string{}
+	if !exact && kd.isFederationQuery(path) {
```

Comment:
I am not sure about this exact parameter.
Looking at the existing code, it seems that we should be checking !exact here.
Can remove it if its not required.